vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -279,57 +305,109 @@ public void add(List<Document> documents) {
 		UploadRequest upload = new UploadRequest(documents.stream().map(document -> {
 			// Compute and assign an embedding to the document.
 			document.setEmbedding(this.embeddingClient.embed(document));
-			return new UploadRequest.Embedding(document.getId(), document.getEmbedding(), documentField,
-					document.getContent(), document.getMetadata());
+			List<Float> floatVector = document.getEmbedding().stream().map(Double::floatValue).toList();
+			return new UploadRequest.Embedding(document.getId(), floatVector, documentField, document.getContent(),
+					document.getMetadata());
 		}).toList());
 
-		client.put()
-			.uri("/keys")
-			.contentType(MediaType.APPLICATION_JSON)
-			.bodyValue(upload)
-			.retrieve()
-			.bodyToMono(Void.class)
-			.block();
+		ObjectMapper objectMapper = new ObjectMapper();
+		try {
+			String embeddingsJson = objectMapper.writeValueAsString(upload).substring("{\"embeddings\":".length());
+
+			client.post()
+				.uri("/" + INDEX_NAME + "/embeddings")
+				.contentType(MediaType.APPLICATION_JSON)
+				.bodyValue(embeddingsJson)
+				.retrieve()
+				.bodyToMono(Void.class)
+				.block();
+		}
+		catch (WebClientResponseException.NotFound notFoundException) {
+			throw new RuntimeException("Vector index does not exist", notFoundException);
+		}
+		catch (WebClientResponseException.BadRequest badRequestException) {
+			throw new RuntimeException("Bad request. The request body could not be processed successfully.",
+					badRequestException);
+		}
+		catch (Exception e) {
+			throw new RuntimeException("An unexpected error occurred", e);
+		}
 	}
 
 	@Override
 	public Optional<Boolean> delete(List<String> idList) {
-		throw new UnsupportedOperationException("This vector store doesn't support delete (yet)");
+		try {
+			client.method(HttpMethod.DELETE)
+				.uri("/" + INDEX_NAME + "/embeddings/")
+				.body(BodyInserters.fromValue(idList))
+				.retrieve()
+				.bodyToMono(Void.class)
+				.block();
+		}
+		catch (Exception e) {
+			throw new RuntimeException("Error removing embedding " + e);
```

Comment:
Not knowing much about the api at the moment, is it expected to throw these exceptinos or handle them and return a false?